graylog2-web-interface/src/components/content-packs/ContentPacksList.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+
+import {
+  Col,
+  Row,
+} from 'components/bootstrap';
+import {
+  Pagination, PageSizeSelect, NoSearchResult, NoEntitiesExist,
+} from 'components/common';
+import TypeAheadDataFilter from 'components/common/TypeAheadDataFilter';
+import ControlledTableList from 'components/common/ControlledTableList';
+import ContentPackListItem from 'components/content-packs/components/ContentPackListItem';
+import { DEFAULT_PAGINATION } from 'stores/PaginationTypes';
+
+import type { ContentPackInstallation, ContentPackMetadata } from './Types';
+
+type Props = {
+  contentPacks: Array<ContentPackInstallation>,
+  contentPackMetadata: ContentPackMetadata,
+  onDeletePack: (id: string) => void,
+  onInstall: (id: string, contentPackRev: string, parameters: unknown) => void,
+};
+
+const StyledPageSizeSelect = styled(PageSizeSelect)(({ theme }) => css`
+  display: flex;
+  align-items: center;
+  gap: ${theme.spacings.xs};
+  float: right;
+`);
+
+const ContentPacksList = ({ contentPacks, contentPackMetadata, onDeletePack, onInstall }: Props) => {
+  const [filteredContentPacks, setFilteredContentPacks] = useState(contentPacks);
+  const [paginationOption, setPaginationOption] = useState(DEFAULT_PAGINATION);
+
+  const formatItems = (items: Array<ContentPackInstallation>) => {
+    const { perPage, page } = paginationOption;
+    const begin = (perPage * (page - 1));
+    const end = begin + perPage;
+    const shownItems = items.slice(begin, end);
+
+    return shownItems.map((item) => (
+      <ContentPackListItem key={item.id}
+                           pack={item}
+                           contentPackMetadata={contentPackMetadata}
+                           onDeletePack={onDeletePack}
+                           onInstall={onInstall} />
+    ));
+  };
+
+  const filterContentPacks = (filteredItems: Array<ContentPackInstallation>) => {
+    setFilteredContentPacks(filteredItems);
+  };
+
+  const itemsShownChange = (pageSize: number) => {
```

Comment:
Absolute nit, but we could try to stick to one naming pattern for handlers, e.g. call this one `onChangeItemsShown` so it is similar to `onChangePage`